### PR TITLE
Convert depreciation plant ID from pudl to eia

### PR DESCRIPTION
**Pull request by cmgosnell**
The main work of this PR is to convert the depreciation related data processing to use the EIA plant ID instead of the PUDL plant ID. This included a ton of work to actually convert the `deprish_raw.xlsx` file to have the EIA ids instead of PUDL ids and converting the few places where we are storing IDs that are generated from these IDs (like the training data in `inputs/train_ferc1_eia.csv` and the common labeling in `inputs/deprish_common_labels.xlsx`)... but most of the code is little changes to work with the EIA id's where appropriate across the pipeline.

- `train_ferc1_eia.csv`: some 600+ lines of this PR is from training data. Mostly this was adding Duke's new mannually mapped training data. You can basically skip this.
- the notebooks!: mostly there are little bb changes to keep them up to date. I'd suggest ignoring them.
- `connect_deprish_to_eia.py`: mostly just `plant_id_pudl` to `plant_id_eia` conversion. Also change mul (master unit list) to ppl (plant part list).
- `connect_deprish_to_ferc1.py`: also mostly just `plant_id_pudl` to `plant_id_eia` conversion.
- `connect_ferc1_to_eia`: I merged some work from @aesharpe 's `mega-ferc-to-eia` branch got merged in here. This included converting to using the `pudl_out.all_plants_ferc1()` instead of just the steam table in the FERC-EIA ml connection module. There is also additional logging for the outputs of the ml model. This is where ~250 of the line changes are.
- `deprish.py `: again, also entirely `plant_id_pudl` to `plant_id_eia` conversion.

TODO for CG:
- [ ] I found a number of depreciation records that need a manual scrub of their plant_id_eia's. Once these ID's are all settled, I'll fully remove the plant_id_pudl column from the raw depreciation sheet.

### CODE REVIEW

File: src/pudl_rmi/coordinate.py [hunk 1]
```diff
@@ -256,7 +256,7 @@ def grab_all(self, clobber_all=False):
             pandas.DataFrame: a table of the conneciton between the
                 depreciation studies and the FERC1 plants.
         """
-        ppl = self.grab_plant_part_list(clobber=False)
+        ppl = self.grab_plant_part_list(clobber=clobber_all)
```
Comment (aesharpe) [inline src/pudl_rmi/coordinate.py:259]: Went from `clobber=False` to `clobber=True`, just checking in why?

File: src/pudl_rmi/deprish.py [hunk 2]
```diff
@@ -30,14 +30,14 @@
 logger = logging.getLogger(__name__)
 
 
-INT_IDS = ['utility_id_ferc1', 'utility_id_pudl',
+INT_IDS = ['utility_id_ferc1', 'utility_id_pudl', 'plant_id_eia',
            'plant_id_pudl', 'report_year']
```
Comment (aesharpe) [inline src/pudl_rmi/deprish.py:34]: Is this supposed to have both `plant_id_eia` and `plant_id_pudl`?

File: src/pudl_rmi/deprish.py [hunk 3]
```diff
@@ -1014,13 +1018,13 @@ def assign_line_id(df):
...
-        x.plant_part_name.map(str).str.lower() + "_" +
-        x.ferc_acct_name.fillna("").str.lower() + "_" +
-        # x.note.fillna("") + "_" +
-        x.utility_id_pudl.map(str) + "_" +
-        x.data_source.fillna("")
+            x.report_date.dt.year.astype(pd.Int64Dtype()).map(str) + "_" +
+            x.plant_id_eia.map(str) + "_" +
+            x.plant_part_name.map(str).str.lower() + "_" +
+            x.ferc_acct_name.fillna("").str.lower() + "_" +
+            # x.note.fillna("") + "_" +
```
Comment (aesharpe) [inline src/pudl_rmi/deprish.py:1025]: Get rid of this commented out line if it's not useful.

File: src/pudl_rmi/deprish.py [hunk 4]
```diff
@@ -1178,10 +1182,10 @@ def get_plant_pudl_info(pudl_out):
         pudl_out.gens_eia860()
         .assign(count='place_holder')
         .sort_values(['plant_name_eia', 'state'])
-        .groupby(['plant_id_pudl', 'report_date'], as_index=False)
+        .groupby(['plant_id_eia', 'report_date'], as_index=False)
         # Must use .join because x.unique() arrays are not hashable
         .agg(
-            {'plant_id_eia':
+            {'plant_id_pudl':
```
Comment (aesharpe) [inline src/pudl_rmi/deprish.py:1188]: Most of the time you're swapping from `plant_id_pudl` to `plant_id_eia` but here you are doing the opposite? Just checking that was intentional

**Review by aesharpe — COMMENTED** [review]
Most of my comments are sanity checks where I wasn't totally sure what was going on, but I flagged something that looked different. 

I would recommend doing a quick search of comments and docstrings to make sure all references to `plant_id_eia` and `plant_id_pudl` are correct there as well as in the actual code. 

There were some other subtle changes that I wasn't really capable of evaluating, like changes to variables or parameters, but I assume that if the code runs then these were part of intentional other changes or improvements made along the way.

File: src/pudl_rmi/connect_deprish_to_eia.py [hunk 5]
```diff
@@ -201,13 +215,13 @@ def match_merge(deprish_df, mul_df, key_deprish, key_mul):
             threshold=75),
         mul_df.drop_duplicates(
             subset=['report_year', 'plant_name_new'])[PPL_COLS],
-        left_on=['report_year', 'utility_id_pudl', 'plant_name_match'],
-        right_on=['report_year', 'utility_id_pudl', key_mul], how='left')
+        left_on=['report_year', 'utility_id_pudl', 'plant_name_match', 'plant_id_eia'],
+        right_on=['report_year', 'utility_id_pudl', key_mul, 'plant_id_eia'], how='left')
```
[the remark this answers is not in the document]
Comment (cmgosnell) [reply src/pudl_rmi/connect_deprish_to_eia.py:219]: I'm honestly not sure why this wasn't here in the first place. It was assuming that the string names were unique enough to merge on. But I found a few instances of plants with the exact same string name but different IDs.

File: src/pudl_rmi/connect_deprish_to_eia.py [hunk 6]
```diff
@@ -282,15 +300,19 @@ def match_deprish_eia(plant_parts_df, sheet_name_output):
     deprish_match = deprish_match.loc[
         :, first_cols + [x for x in deprish_match.columns
                          if x not in first_cols]]
+    return deprish_match
+
 
+def grab_possible_plant_part_list_matches(plant_parts_df, deprish_df):
+    "Docs."
```
[the remark this answers is not in the document]
Comment (cmgosnell) [reply src/pudl_rmi/connect_deprish_to_eia.py:307]: lol yes! ty!

File: src/pudl_rmi/coordinate.py (hunk 1 above)
Comment (cmgosnell) [reply src/pudl_rmi/coordinate.py:259]: ah yes. I had left a dangling false in there for the ppl bc it takes forever and I was testing everything else. but this should be `clobber_all`

File: src/pudl_rmi/deprish.py (hunk 2 above)
Comment (cmgosnell) [reply src/pudl_rmi/deprish.py:34]: good catch! I had both of these in here for a bit when I had both ID's in the raw depreciation file, but I believe I can remove this now bc I've removed the pudl id.

File: src/pudl_rmi/connect_deprish_to_ferc1.py [hunk 7]
```diff
@@ -23,9 +23,11 @@
 
 logger = logging.getLogger(__name__)
 
-CONNECT_COLS = ['plant_id_pudl',
-                'utility_id_pudl',
-                'report_date']
+CONNECT_COLS = [
+    'plant_id_pudl',
+    # 'utility_id_pudl',
```
[the remark this answers is not in the document]
Comment (cmgosnell) [reply src/pudl_rmi/connect_deprish_to_ferc1.py:28]: I have completely revamped this module in PR #150. I appreciate these comments in here, but I am going to mark them all as resolved bc I effectively replaced this whole guy.

File: src/pudl_rmi/connect_deprish_to_eia.py [hunk 8]
```diff
@@ -96,6 +96,20 @@ def prep_deprish(plant_parts_df, key_deprish):
         on=RESTRICT_MATCH_COLS,
         validate='m:1'
     )
+    # check the number of depreciation records that should have EIA plant-part
+    # matches.
+    # TODO: go through all of these to reassign plant_id_eia!!! and turn down
+    # the acceptable number of baddies
```
[the remark this answers is not in the document]
Comment (cmgosnell) [reply src/pudl_rmi/connect_deprish_to_eia.py:102]: this is actually not done :-( I needed to move ahead due to time constraints. I'll make a new little issue specifically for this.

File: src/pudl_rmi/deprish.py (hunk 4 above)
Comment (cmgosnell) [reply src/pudl_rmi/deprish.py:1188]: this is a good question! but yes in this case it is intentional. I am grouping by the plant_id_eia above so I effectively swapped them (one in the groupby one in the agg).

**Pull request merged**